File: android/app/src/main/kotlin/com/actduck/videogame/data/AdAnalytics.kt
package com.actduck.videogame.data

import androidx.room.Entity
import androidx.room.PrimaryKey
// import com.actduck.libad.DuckAds.AdFormat
import java.io.Serializable
import java.util.Date

@Entity
class AdAnalytics(
  @PrimaryKey var id: Long? = null,
  var userId: Long? = null,
  var userPseudoId: String? = null,
  var userIP: String? = null,
  var adFormat: Int? = null,
  var event: Int? = null, // 显示还是点击 1 显示 2 点击
  var date: Date? = null,
  var uploaded: Boolean = false // 已上传服务器
) : Serializable {

}

//{ "clickTimes": 3, "enable": true, "interval": 60000 }
data class AdsConfig(
  val clickTimes: Int,
  val enable: Boolean,
  val interval: Long
)

File: lib/model/ad_analytics.dart
class AdAnalytics {
  int? id = null;
  int? userId = null;
  String? userPseudoId = null;
  String? userIP = null;
  int? adFormat = null;
  int? event = null; // 显示还是点击 1 显示 2 点击
  DateTime? date = null;
  bool? uploaded = false; // 已上传服务器

  factory AdAnalytics.fromMap(Map<String, dynamic> json) {
    return AdAnalytics(
      id: json["id"],
      userId: json["userId"],
      userPseudoId: json["userPseudoId"],
      userIP: json["userIP"],
      adFormat: json["adFormat"],
      event: json["event"],
      date: DateTime.parse(json["date"]),
      uploaded: json["uploaded"] == 1,
    );
  }

  Map<String, dynamic> toMap() => {
        "id": id,
        "userId": userId,
        "userPseudoId": userPseudoId,
        "userIP": userIP,
        "adFormat": adFormat,
        "event": event,
        "date": date.toString(),
        "uploaded": uploaded,
      };

  AdAnalytics(
      {this.id, this.userId, this.userPseudoId, this.userIP, this.adFormat, this.event, this.date, this.uploaded});

  factory AdAnalytics.ad(int adFormat, int event) => AdAnalytics(adFormat: adFormat, event: event, uploaded: false);
}
